src/yoroi-wallets/storage.ts

```diff
@@ -1,46 +1,62 @@
-import AsyncStorage, {AsyncStorageStatic} from '@react-native-async-storage/async-storage'
-
-export type Storage = {
-  getItem: AsyncStorageStatic['getItem']
-  multiGet: AsyncStorageStatic['multiGet']
-  setItem: AsyncStorageStatic['setItem']
-  multiSet: AsyncStorageStatic['multiSet']
-  removeItem: AsyncStorageStatic['removeItem']
-  multiRemove: AsyncStorageStatic['multiRemove']
-  clear: AsyncStorageStatic['clear']
-}
+import AsyncStorage from '@react-native-async-storage/async-storage'
+
+import {parseSafe} from './utils/parsing'
+
+export type Storage = ReturnType<typeof mountStorage>
+export type Path = `${string}/`
+
+const mountStorage = (path: Path) => {
+  const withPath = (key: string) => `${path}${key}` as `${Path}${string}`
+  const withoutPath = (value: string) => value.slice(path.length)
+
+  function getItem<T>(key: string, parse: (item: string | null) => T): Promise<T>
+  function getItem<T = unknown>(key: string): Promise<T>
+  async function getItem(key: string, parse = parseSafe) {
+    const item = await AsyncStorage.getItem(withPath(key))
+    return parse(item)
+  }
 
-export const makeStorageWithPrefix = (prefix: string): Storage => {
-  const withPrefix = (key: string) => `${prefix}${key}`
-  const withoutPrefix = (value: string) => value.slice(prefix.length)
+  function multiGet<T>(keys: Array<string>, parse: (item: string | null) => T): Promise<Array<[string, T]>>
+  function multiGet<T = unknown>(keys: Array<string>): Promise<Array<[string, T]>>
+  async function multiGet(keys: Array<string>, parse = parseSafe) {
+    const absolutePaths = keys.map((key) => withPath(key))
+    const items = await AsyncStorage.multiGet(absolutePaths)
+    return items.map(([key, value]) => [withoutPath(key), parse(value)] as const)
+  }
 
   return {
-    getItem: (key: string) => {
-      return AsyncStorage.getItem(withPrefix(key))
-    },
-    multiGet: async (keys: Array<string>) => {
-      const prefixedKeys = keys.map((key) => withPrefix(key))
-      const items = await AsyncStorage.multiGet(prefixedKeys)
+    join: (folderName: Path) => mountStorage(`${path}${folderName}`),
```

Comment:
I'm guessing that the `Path` type, dismisses the necessity to test paths ending with`/`.